### PR TITLE
Bug 1915840: Store secrets in one place and utilize mutex

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR is an aggregation of the upstream changes made to certificate logic
* Moves stashing of generated certs to the "master-certs" secret
* Corrects the cert extraction logic to extract all certs from the master-cert secret
* Reformats the cert gen script output to be parsable to allow explicit understandying of why certs were regenerated
* Adds a test to expose the issue
* Writing the secret when it changes or when it was not found initially

/cc @anpingli @QiaolingTang 
/assign @ewolinetz @vimalk78 

### Links
4.5 backports of #775, #849

https://bugzilla.redhat.com/show_bug.cgi?id=1915840
https://bugzilla.redhat.com/show_bug.cgi?id=1906641
https://bugzilla.redhat.com/show_bug.cgi?id=1901869

### CODE REVIEW

File: pkg/k8shandler/certificates_test.go [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+			Expect(secret).Should(SucceedVerifyX509("ca.crt", "ca.crt", "openshift-cluster-logging-signer", nil, nil, nil))
+		})
+
+		It("skip re-generating secrets during restart", func() {
+			secret := &corev1.Secret{}
+			key := types.NamespacedName{Name: constants.MasterCASecretName, Namespace: constants.OpenshiftNS}
+			Expect(client.Get(context.TODO(), key, secret)).Should(Succeed())
+
+			Expect(os.RemoveAll(utils.GetWorkingDir())).To(Succeed())
+			clusterRequest.CreateOrUpdateCertificates()
```
Comment (vimalk78) [inline pkg/k8shandler/certificates_test.go:134]: a lint error here
`Error return value of `clusterRequest.CreateOrUpdateCertificates` is not checked (errcheck)`

**Comment by jcantrill** [comment]
Overriding test as history shows it is failing consistently on the same test:
* Managed ES LF test which uses the changes associated with PR consistently succeeds
* Unmanaged ES LF test shows evidence of being able to collect infra, audit log; does not use this change in production
* All other tests consistently pass

**Pull request merged**